evm/test/Upgrades.t.sol

```diff
@@ -184,6 +184,49 @@ contract TestUpgrades is Test, IRateLimiterEvents {
         basicFunctionality();
     }
 
+    function test_cannotUpgradeToNoRateLimitingIfItWasEnabled() public {
```

Comment:
Upgrading from rate limiting enabled to the new `NoRateLimiting` variant is technically possible by migrating the immutables. Any queued transfers would be stuck but otherwise I think it would work fine. Maybe worth adding a warning to the README or somewhere else strongly discouraging this upgrade?

The counter case would be deploying a `NoRateLimiting` setup initially and then upgrading to enable rate limiting in the future, which I think would be possible if the immutables were migrated to pass the immutables check. This seems like a perfectly fine thing to do and could maybe do with a test? Similarly maybe worth adding a line to the docs

Reply:
Agree, it is possible, but as you mention, the integrator would need to modify this contract to allow releasing queued transfers (but we were trying to entirely drop that code). I'd agree with adding a readme note on migrating from `NttManager` to NttManagerNoRateLimiting` with this restriction.

Also agree providing a case of an `NttManagerNoRateLimiting` upgrading to an `NttManager` should work, though as you mention the integrator would need to purposefully change the immutables check if they actually wanted it to be enabled.